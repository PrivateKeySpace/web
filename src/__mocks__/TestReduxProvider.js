import React from 'react'
import PropTypes from 'prop-types'
import { createStore } from 'redux'
import { Provider as ReduxProvider } from 'react-redux'
import { compose, setDisplayName, setPropTypes, defaultProps } from 'recompose'
import rootReducer from '../store/rootReducer'

const displayName = '__mocks__/TestReduxProvider'

const enhance = compose(
  setDisplayName(displayName),
  setPropTypes({ state: PropTypes.object, children: PropTypes.any }),
  defaultProps({ state: {} })
)

function TestReduxProvider (props) {
  const { state, children } = props
  const store = createStore(rootReducer, state)

  return (
    <ReduxProvider store={store}>
      {children}
    </ReduxProvider>
  )
}

export default enhance(TestReduxProvider)
