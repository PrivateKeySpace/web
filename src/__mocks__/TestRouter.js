import React from 'react'
import PropTypes from 'prop-types'
import { MemoryRouter } from 'react-router'
import { compose, setDisplayName, setPropTypes, defaultProps } from 'recompose'

const displayName = '__mocks__/TestRouter'

const enhance = compose(
  setDisplayName(displayName),
  setPropTypes({ path: PropTypes.string, children: PropTypes.any }),
  defaultProps({ path: '/' })
)

function TestRouter (props) {
  const { path, children } = props

  return (
    <MemoryRouter initialEntries={[path]} initialIndex={0}>
      {children}
    </MemoryRouter>
  )
}

export default enhance(TestRouter)
