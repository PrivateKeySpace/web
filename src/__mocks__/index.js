export { default as TestRouter } from './TestRouter'
export { default as TestReduxProvider } from './TestReduxProvider'
