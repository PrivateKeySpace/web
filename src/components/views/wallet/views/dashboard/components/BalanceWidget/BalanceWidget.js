import React from 'react'
import { compose, setDisplayName, setPropTypes, defaultProps, pure } from 'recompose'
import { css } from 'aphrodite/no-important'
import { propTypes, defaultPropsSpec } from './lib'
import tx from '../../../../../../../theme/styles'
import sx from './styles'
import { Link } from '../../../../../../elements'

const displayName = 'components/views/wallet/views/dashboard/components/BalanceWidget'
const enhance = compose(
  setDisplayName(displayName),
  setPropTypes(propTypes),
  defaultProps(defaultPropsSpec),
  pure
)

function BalanceWidget (props) {
  const { title, icon, link, usdValue, btcValue } = props

  return (
    <div className={css(tx.mr2, tx.flex, tx.flexDirectionRow)}>
      {
        icon &&
        <div className={css(tx.mr4, tx.pt1, sx.icon)}>
          <span className={`lnr lnr-${icon}`} />
        </div>
      }
      <div className={css(sx.info)}>
        <div className={css(tx.textTransformUppercase, tx.fontWeight400, sx.accountName)}>{title}</div>
        <div className={css(tx.flex, tx.flexDirectionRow, tx.alignItemsCenter, sx.balance)}>
          <div className={css(tx.fontWeight400, sx.usdBalance)}>{usdValue} USD</div>
          <div className={css(tx.ml2, sx.btcBalance)}>
            {btcValue}
            <br />
            BTC
          </div>
        </div>
        {
          link &&
          <Link to={link.to}>{link.text}</Link>
        }
      </div>
    </div>
  )
}

export default enhance(BalanceWidget)
