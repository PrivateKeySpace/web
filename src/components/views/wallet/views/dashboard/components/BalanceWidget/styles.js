import { StyleSheet } from 'aphrodite/no-important'
import { colors } from '../../../../../../../theme/constants'

const styles = StyleSheet.create({
  icon: {
    fontSize: 30,
    color: colors.gray
  },
  accountName: {
    fontSize: 12,
    color: colors.gray
  },
  usdBalance: {
    fontSize: 30,
    color: colors.black
  },
  btcBalance: {
    fontSize: 12,
    color: colors.gray
  }
})

export default styles
