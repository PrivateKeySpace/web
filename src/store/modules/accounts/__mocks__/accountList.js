const accountList = [
  {
    'path': [
      2147483697,
      2147483648,
      2147483648
    ],
    'address': '3Pc8vh9xUopMmiBW1yDAm3h1aqqbmmv11p',
    'addressPath': [
      2147483697,
      2147483648,
      2147483648,
      0,
      0
    ],
    'addressId': 0,
    'xpub': 'xpub6Ca3FEzN1SemuKNujpAeyDLXk7DXXtgpUs5HHLrG5brjVQm7ib3SLqMKfWWQZset7HMm7wM58wVGEJm5eG6rkxEcYvtvExYaM9T7jxkmmMp',
    'balance': 0,
    'confirmed': 0,
    'id': 0,
    'segwit': true
  },
  {
    'path': [
      2147483692,
      2147483648,
      2147483648
    ],
    'address': '1Bswvjfoa1p1kMkwE4C8KAPbYLSKfg1Nni',
    'addressPath': [
      2147483692,
      2147483648,
      2147483648,
      0,
      0
    ],
    'addressId': 0,
    'xpub': 'xpub6D4zdJZuWGKQK5qMyj7reDcapRE6StaVTGS3xWCAjRyLfne84tGxUK1yYNrUWFY8gYtJA7xd3s97o1uiPkFEhpdsuDsNkLnj5iHyAbCEnRv',
    'balance': 0,
    'confirmed': 0,
    'id': 0,
    'segwit': false
  }
]

export default accountList
