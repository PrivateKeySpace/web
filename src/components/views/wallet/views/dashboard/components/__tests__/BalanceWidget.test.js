import React from 'react'
import { StyleSheetTestUtils } from 'aphrodite'
import { render } from 'enzyme'
import BalanceWidget from '../BalanceWidget'
import { TestRouter } from '../../../../../../../__mocks__'

beforeEach(() => {
  StyleSheetTestUtils.suppressStyleInjection()
})

afterEach(() => {
  StyleSheetTestUtils.clearBufferAndResumeStyleInjection()
})

it('renders without crashing and matches snapshot', () => {
  const widget = render(
    <TestRouter>
      <BalanceWidget title='Test title' icon='briefcase' link={{ to: '/test', text: 'Test Link' }} />
    </TestRouter>
  )
  expect(widget).toMatchSnapshot()
})

it('renders without icon, does not crash and matches snapshot', () => {
  const widget = render(
    <TestRouter>
      <BalanceWidget title='Test title' link={{ to: '/test', text: 'Test Link' }} />
    </TestRouter>
  )
  expect(widget).toMatchSnapshot()
})

it('renders without icon and without link, does not crash and matches snapshot', () => {
  const widget = render(
    <TestRouter>
      <BalanceWidget title='Test title' />
    </TestRouter>
  )
  expect(widget).toMatchSnapshot()
})
