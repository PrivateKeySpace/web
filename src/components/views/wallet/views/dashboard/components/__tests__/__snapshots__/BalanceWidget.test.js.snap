// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing and matches snapshot 1`] = `
<div
  class="mr2_18emzam-o_O-flex_jro6t0-o_O-flexDirectionRow_qb00m9"
>
  <div
    class="mr4_kh8dwt-o_O-pt1_vsuzfi-o_O-icon_nkv9vo"
  >
    <span
      class="lnr lnr-briefcase"
    >
      
    </span>
  </div>
  <div
    class=""
  >
    <div
      class="textTransformUppercase_4efw5a-o_O-fontWeight400_tt122m-o_O-accountName_1yz1w90"
    >
      Test title
    </div>
    <div
      class="flex_jro6t0-o_O-flexDirectionRow_qb00m9-o_O-alignItemsCenter_1moixrt"
    >
      <div
        class="fontWeight400_tt122m-o_O-usdBalance_1fhummc"
      >
        0 USD
      </div>
      <div
        class="ml2_y4bb05-o_O-btcBalance_1yz1w90"
      >
        0
        <br>
          
        </br>
        BTC
      </div>
    </div>
    <a
      class="textTransformUppercase_4efw5a-o_O-fontWeight400_tt122m-o_O-fontSize3_o9hx2m-o_O-link_1onilyj"
      href="/test"
    >
      Test Link
    </a>
  </div>
</div>
`;

exports[`renders without icon and without link, does not crash and matches snapshot 1`] = `
<div
  class="mr2_18emzam-o_O-flex_jro6t0-o_O-flexDirectionRow_qb00m9"
>
  <div
    class=""
  >
    <div
      class="textTransformUppercase_4efw5a-o_O-fontWeight400_tt122m-o_O-accountName_1yz1w90"
    >
      Test title
    </div>
    <div
      class="flex_jro6t0-o_O-flexDirectionRow_qb00m9-o_O-alignItemsCenter_1moixrt"
    >
      <div
        class="fontWeight400_tt122m-o_O-usdBalance_1fhummc"
      >
        0 USD
      </div>
      <div
        class="ml2_y4bb05-o_O-btcBalance_1yz1w90"
      >
        0
        <br>
          
        </br>
        BTC
      </div>
    </div>
  </div>
</div>
`;

exports[`renders without icon, does not crash and matches snapshot 1`] = `
<div
  class="mr2_18emzam-o_O-flex_jro6t0-o_O-flexDirectionRow_qb00m9"
>
  <div
    class=""
  >
    <div
      class="textTransformUppercase_4efw5a-o_O-fontWeight400_tt122m-o_O-accountName_1yz1w90"
    >
      Test title
    </div>
    <div
      class="flex_jro6t0-o_O-flexDirectionRow_qb00m9-o_O-alignItemsCenter_1moixrt"
    >
      <div
        class="fontWeight400_tt122m-o_O-usdBalance_1fhummc"
      >
        0 USD
      </div>
      <div
        class="ml2_y4bb05-o_O-btcBalance_1yz1w90"
      >
        0
        <br>
          
        </br>
        BTC
      </div>
    </div>
    <a
      class="textTransformUppercase_4efw5a-o_O-fontWeight400_tt122m-o_O-fontSize3_o9hx2m-o_O-link_1onilyj"
      href="/test"
    >
      Test Link
    </a>
  </div>
</div>
`;
